homeassistant/components/ads/sensor.py

```diff
@@ -17,13 +24,16 @@
     {
         vol.Required(CONF_ADS_VAR): cv.string,
         vol.Optional(CONF_ADS_FACTOR): cv.positive_int,
+        vol.Optional(CONF_ADS_DIGIT, default=ads.DIGIT_DEFAULT): cv.positive_int,
```

Comment:
We're changing the platform config here. We want all platform config to be hosted in the main integration config.

We may need to refactor this before we can accept this change.

See https://github.com/home-assistant/architecture/blob/master/adr/0007-integration-config-yaml-structure.md

Reply:
I understand, I think that's good if there is only one configuration option left.

As I told you before, I haven't had much experience with Python. Is there already an integration that implements this, e.g. KNX integration? It will certainly help me if I can read the code there.

For you as information. I would like to further expand ADS integration in the next few months. I would add String as the next data type.

I would like to program short and long commands for the blinds, similar to the KNX.

With light support for tunable white, RGBW etc.

There is also a problem with the ADS Hub. As long as it is running, everything is fine, but if the connection to the PLC is lost, HA no longer automatically establishes it. You have to restart the HA manually so that it re-establishes the connection.

I would like to install such things in the next few months. Since I just don't have that much Python experience, at first only small and light things like data types and with more experience the increasingly "complex" things

Reply:
The `netgear_lte` integration loads its platforms via discovery with `async_load_platform`.